storage/inkless/src/main/java/io/aiven/inkless/common/SharedState.java

```diff
@@ -74,13 +112,58 @@ public static SharedState initialize(
         );
     }
 
+    public StorageBackend storage() {
```

Comment:
Each call to get the storage will create new `StorageBackend` object. `FetchHandler`, `FileCleaner`, `FileMerger` and others will have their own storage backend object which initializes quite a lot of resources.

The `SharedState` should be a singleton and immutable and I think it would be best to return a one fixed storage backend object?